core/src/cluster_info.rs

```diff
@@ -763,6 +777,31 @@ impl ClusterInfo {
 
         Ok((addr, out))
     }
+    // if the network entry point hasn't been discovered yet, add it to the crds table
+    fn add_entrypoint(&mut self, pulls: &mut Vec<(Pubkey, Bloom<Hash>, SocketAddr, CrdsValue)>) {
```

Comment:
I don't understand this function.
1. new_pull_requests creates a new pull request
2. if the new pull request is empty, use self.entrypoint
3 goto 1

Reply:
The intention was 

1. New pull requests creates a new pull
2. If the entrypoint is some and is not in crds, try to also send a pull to the entrypoint, else unset it
3. Goto 1

You think doing it only if the pulls are empty is sufficient?

Reply:
@aeyakovenko, yeah you're right, I only care about this if I have no pull requests. Since it's the _entrypoint_ if we have valid pulls, I can just discard the entrypoint. I'll update the PR to reflect this